File: wrappers/nodejs/examples/overhead.js
const { NanoChronometer } = require('..');
const nc = new NanoChronometer();
nc.calibrate(50);
console.log('native overhead cycles:', nc.nativeOverheadCycles().toString());
console.log('Node -> native ffi-napi cycles:', nc.ffiOverheadCycles(10000).toString());
console.log('native -> JS callback min cycles:', nc.callbackMinCycles(() => {}, 10000).toString());
nc.close();

File: wrappers/zig/overhead.zig
const std = @import("std");
const ncmod = @import("nanochronometer.zig");

pub fn main() !void {
    var nc = try ncmod.NanoChronometer.init();
    defer nc.deinit();
    _ = nc.calibrate(50);
    std.debug.print("native overhead cycles: {d}\n", .{nc.nativeOverheadCycles()});
    std.debug.print("Zig -> native cycles: {d}\n", .{nc.ffiOverheadCycles(10000)});
}
